src/interfaces/IAuthorizedEvent.ts

```diff
@@ -3,41 +3,49 @@ export type IAuthorizedEvent<TokenPayload = any> =
   | ILowerCaseAuthorizedEvent<TokenPayload>
   | IUpperCaseAuthorizedEvent<TokenPayload>
 
-export interface IAuthorizedEventBase<TokenPayload = any> {
+export type IAuthorizedEventBase<TokenPayload = any> = {
   /** Authorization information added by this middleware from a JWT. Has to be undefined before hitting the middleware. */
   auth?: {
     payload: TokenPayload
     token: string
   }
   /** An object containing event headers */
   headers: any
-  /** The http request method of this event */
-  httpMethod: string
-}
+} & (
+  | {
+      /** The http request method of this event (for REST APIs) */
+      httpMethod: string
+    }
+  | {
+      /** The metadata about this event (for HTTP APIs) */
+      requestContext: {
+        /** The http request metadata about this event */
+        http: object
+      }
+    }
+)
 
 /** An event with a lower case authorization header */
-export interface ILowerCaseAuthorizedEvent<TokenPayload = any>
-  extends IAuthorizedEventBase<TokenPayload> {
+export type ILowerCaseAuthorizedEvent<TokenPayload = any> = {
   headers: {
     /**
      * The authorization token to check. Can be a string or an array with exactly one string.
      * @example "Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiaWF0IjoxNTE2MjM5MDIyfQ.SflKxwRJSMeKKF2QT4fwpMeJf36POk6yJV_adQssw5c"
      */
     authorization: string | string[]
   }
-}
+} & IAuthorizedEventBase<TokenPayload>
```

Comment:
Please stay with interfaces where possible as these are parsed faster by TypeScript.

Reply:
I've used interfaces where possible, but I believe these still need to be types as it is not possible to have a union within an interface. Here, `IAuthorizedApiGatewayEvent` can either be a `IAuthorizedRestApiGatewayEvent` or a `IAuthorizedHttpApiGatewayEvent`